pkg/validations/upgradevalidations/versions_test.go

```diff
@@ -43,6 +43,18 @@ func TestValidateVersionSkew(t *testing.T) {
 			upgradeVersion:        v1alpha1.Kube120,
 			serverVersionResponse: "testdata/kubectl_version_server_119.json",
 		},
+		{
+			name:                  "FailureParsingVersion",
+			wantErr:               errors.New("parsing comparison version: could not parse \"test\" as version"),
+			upgradeVersion:        "test",
+			serverVersionResponse: "testdata/kubectl_version_server_119.json",
+		},
+		{
+			name:                  "FailureParsingServerVersion",
+			wantErr:               errors.New("parsing cluster version: could not parse \"test\" as version"),
+			upgradeVersion:        v1alpha1.Kube119,
+			serverVersionResponse: "testdata/kubectl_invalid_server.json",
```

Comment:
I'm very confused by this
Why are we passing jsons to the test and why are we adding new test cases for invalid json?

Reply:
The invalid test case was added to cover the parsing cluster version error for CLI specifically. This is not entirely necessary because we usually assume valid kube version coming in from the cluster.

As for why we pass jsons is I think to mock the response received from the server. This use case is covered in the unit test where the main logic resides so this extra one can be removed.